packages/core/src/external/aws/lambda-logic/bundle-to-brief.ts

```diff
@@ -52,117 +59,205 @@ const referenceResources = [
   "Location",
 ];
 
+const UNANSWERED_CALL = "unanswered call";
+const SCHEDULING_CALL = "scheduling call";
+const ADMIN_NOTE = "admin note";
+const SCAN_REF_NOTE = "scan reference";
+
+const REPORT_TYPES_BLACKLIST = [
+  "instructions",
+  "addendum",
+  "nursing note",
+  SCHEDULING_CALL,
+  UNANSWERED_CALL,
+  ADMIN_NOTE,
+  SCAN_REF_NOTE,
+];
+
+const REMOVE_FROM_NOTE = [
+  "xLabel",
+  "5/5",
+  "Â°F",
+  "Â",
+  "â¢",
+  "documented in this encounter",
+  "xnoIndent",
+  "Formatting of this note might be different from the original.",
+  "Formatting of this note is different from the original.",
+  "Portions of the history and exam were entered using voice recognition software",
+  "Images from the original note were not included.",
+  "Minor syntax, contextual, and spelling errors may be related to the use of this software and were not intentional. If corrections are necessary, please contact provider.",
+  "<content>",
+  "</content>",
+  "<root>",
+  "</root>",
+  "&lt;",
+  "&gt;",
+];
+
+const documentVariableName = "text";
+
 export type Brief = {
   id: string;
   content: string;
   link: string;
 };
-
 //--------------------------------
 // AI-based brief generation
 //--------------------------------
 export async function summarizeFilteredBundleWithAI(
-  bundle: Bundle<Resource>
+  bundle: Bundle<Resource>,
+  cxId: string,
+  patientId: string
 ): Promise<string | undefined> {
+  const requestId = uuidv7();
+  const startedAt = new Date();
+  const { log } = out(`summarizeFilteredBundleWithAI - cxId ${cxId}, patientId ${patientId}`);
   // filter out historical data
-  const numHistoricalYears = 2;
-  const dateFrom = dayjs().subtract(numHistoricalYears, "year").format(ISO_DATE);
+  log(`Starting with requestId ${requestId}, and bundle length ${bundle.entry?.length}`);
+  const latestReportDate = findDiagnosticReportResources(bundle)
+    .flatMap(report => {
+      return getDatesFromEffectiveDateTimeOrPeriod(report);
+    })
+    .filter((date): date is string => date !== undefined)
+    .sort((a, b) => b.localeCompare(a))[0];
+  const numHistoricalYears = 1;
+  const initialDate = dayjs(latestReportDate) ?? dayjs();
```

Comment:
let's use the `buildDayJs` function please

Reply:
Fixed!